cli/util.go

```diff
@@ -852,7 +857,11 @@ func doReqAsync(req any, subj string, waitFor int, nc *nats.Conn, cb func([]byte
 			}
 		}
 
-		if finisher != nil {
+		// Set timeout for the next response to arrive.
+		// Unless:
+		//  - We are waiting for a specific number of responses (finisher is null) OR
+		//  - We were told explicitly to wait for the full interval for responses
+		if finisher != nil && waitFor != doReqAsyncWaitTimeoutInterval {
```

Comment:
wouldn't finisher always be nil if waitFor is -1, so just checking `if finisher != nil` should be sufficient here, if not we should restructure things that the nil check is sufficient, easy to miss multiple places to update if this behaviour changes in future again.

Reply:
The default timeout is 5s, so waits 5 extra seconds from the last response. 

I think "stop doing smart things and just wait a fixed amount of time" is a more useful "alternative mode" for this utility function.

Reply:
Still convinced you should just write your own request method. They have different goals.